Counterexamples/IrrationalPowerOfIrrational.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2024 Seewoo Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Seewoo Lee
+-/
+import Mathlib.Data.Real.Irrational
+import Mathlib.Analysis.SpecialFunctions.Pow.NNReal
+
+/-
+# Irrational power of irrational numbers are not necessarily irrational.
+
+Prove that there exist irrational numbers `a`, `b` such that `a^b` is rational.
+
+We use the following famous argument (based on the law of excluded middle and irrationality of √2).
+Consider `c = √2^√2`. If `c` is rational, we are done.
+If `c` is irrational, then `c^√2 = 2` is rational, so we are done.
+-/
+
+
+open Classical Real
+
+namespace Counterexample
+
+/--
+There exist irrational `a`, `b` with rational `a^b`.
+Note that the positivity assumption on `a` is required because of the definition of `rpow` for
+negative bases. See `Real.rpow_def_of_neg` for more details.
```

Comment:
I think "required" is the wrong perspective here, including it just makes the counterexample more interesting.